### PR TITLE
[NEW] - Add setting to request a comment when closing Livechat room

### LINKED ISSUE #12428: [Livechat] Improvement - Deactivation of closing comment
**Is your feature request related to a problem? Please describe.**
When I close a livechat conversation, I always have to insert a closing comment, which frustrates me since we do not use these comments at all.

**Describe the solution you'd like**
As livechat admin I really like to decide wether this comment is optional or not relevant at all. If it is not relevant at all, the comment dialog is not displayed at all. Two solutions
1) make the comment optional
2) do not display it at all

**Additional context**
We implemented the second solution in our branch long time ago: https://github.com/assistify/Rocket.Chat/commit/ab98c8505f8e409f39b28589a8a82886b314b68d

### PR DESCRIPTION
**Pull request by knrt10**
Closes #12428

cc @renatobecker would you please review and add livechat label to it.

### CODE REVIEW

File: app/livechat/client/views/app/tabbar/visitorInfo.js [hunk 1]
```diff
@@ -171,24 +172,40 @@ Template.visitorInfo.events({
...
+						text: t('Chat_closed_successfully'),
+						type: 'success',
+						timer: 1000,
+						showConfirmButton: false,
+					});
+				});
+			});
+		} else {
+			const comment = TAPi18n.__('Chat_closed_by_agent');
+			Meteor.call('livechat:closeRoom', this.rid, comment, function(error/* , result*/) {
```
Comment (renatobecker) [inline app/livechat/client/views/app/tabbar/visitorInfo.js]: This code is duplicated. Why not just creating a const (arrow function) and then pass the comment as a parameter?

File: app/livechat/client/views/app/tabbar/visitorInfo.js [hunk 2]
```diff
@@ -171,24 +172,40 @@ Template.visitorInfo.events({
...
-			type: 'input',
-			inputPlaceholder: t('Please_add_a_comment'),
-			showCancelButton: true,
-			closeOnConfirm: false,
-		}, (inputValue) => {
-			if (!inputValue) {
-				modal.showInputError(t('Please_add_a_comment_to_close_the_room'));
-				return false;
-			}
+		if (settings.get('Livechat_conversation_finished_message_enabled')) {
```
Comment (renatobecker) [inline app/livechat/client/views/app/tabbar/visitorInfo.js]: Please, add a const (arrow function) before this line, and then you can just do something like this:
```
const closeRoom = (comment) => ...here you can call Meteor.call('livechat:closeRoom' ....
if (!settings.get('Livechat_conversation_finished_message_enabled')) {
    return closeRoom(...);
}
... here you can call de modal... closeRoom... and so on..
```

File: app/livechat/server/config.js [hunk 3]
```diff
@@ -80,6 +80,13 @@ Meteor.startup(function() {
 	settings.add('Livechat_allow_switching_departments', true, { type: 'boolean', group: 'Livechat', public: true, i18nLabel: 'Allow_switching_departments' });
 	settings.add('Livechat_show_agent_email', true, { type: 'boolean', group: 'Livechat', public: true, i18nLabel: 'Show_agent_email' });
 
+	settings.add('Livechat_conversation_finished_message_enabled', true, {
```
Comment (renatobecker) [inline app/livechat/server/config.js]:
```suggestion
	settings.add('Livechat_request_comment_when_closing_conversation', true, {
```


request comment when ending live chat conversation

**Comment by knrt10** [comment]
Updated

File: app/livechat/server/config.js (hunk 3 above)
Comment (renatobecker) [reply app/livechat/server/config.js]: @knrt10, once we change the name of the setting, you need to change it's translations string as well.

File: app/livechat/server/config.js [hunk 4]
```diff
@@ -80,6 +80,13 @@ Meteor.startup(function() {
 	settings.add('Livechat_allow_switching_departments', true, { type: 'boolean', group: 'Livechat', public: true, i18nLabel: 'Allow_switching_departments' });
 	settings.add('Livechat_show_agent_email', true, { type: 'boolean', group: 'Livechat', public: true, i18nLabel: 'Show_agent_email' });
 
+	settings.add('Livechat_request_comment_when_closing_conversation', true, {
+		type: 'boolean',
+		group: 'Livechat',
+		public: true,
+		i18nLabel: 'Ask_for_conversation_finished_message',
```
Comment (renatobecker) [inline app/livechat/server/config.js]: You need to change the translation strings too.

File: app/livechat/server/config.js (hunk 3 above)
Comment (knrt10) [reply app/livechat/server/config.js]: ok, let me update

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 5]
```diff
@@ -360,6 +360,7 @@
   "Are_you_sure": "Are you sure?",
   "Are_you_sure_you_want_to_delete_your_account": "Are you sure you want to delete your account?",
   "Are_you_sure_you_want_to_disable_Facebook_integration": "Are you sure you want to disable Facebook integration?",
+  "Livechat_request_comment_when_closing_conversation": "Livechat request comment when closing conversation",
```
Comment (renatobecker) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: - This translation is not in the right order;
- Remove `Livechat` From both string;

File: packages/rocketchat-i18n/i18n/en.i18n.json [hunk 6]
```diff
@@ -567,6 +568,7 @@
   "Chatpal_Timeout_Size_Description": "The time between 2 index windows in ms (on bootstrapping)",
   "Chatpal_Users": "Users",
   "Chatpal_Welcome": "Enjoy your search!",
+  "Chat_closed_by_agent": "Chat closed by agent",
```
Comment (renatobecker) [inline packages/rocketchat-i18n/i18n/en.i18n.json]: - This translation is not in the right order;

**Pull request merged**